### DISCUSSION TITLE
How to debug imported classes

### DISCUSSION THREAD

**Discussion by dudeNumber4** [microsoft/vscode-java-debug#1226] [Q&A] [answered]
I've searched, I can't find a way to debug classes from imported jars.  I can step into imported classes, and I even see local variables load, but the code shown in the editor just has method interfaces defined; no code within any functions/methods.
Am I configured incorrectly?

**Comment by tap90** [comment]
The same problem here but in the preview version of the debugger I was able to navigate the imported classes file of the project

**Reply by dudeNumber4** [reply]
I'm using v0.45.0 preview

**Comment by testforstephen** [comment]
> I've searched, I can't find a way to debug classes from imported jars. I can step into imported classes, and I even see local variables load, but the code shown in the editor just has method interfaces defined; no code within any functions/methods. Am I configured incorrectly?

If the associated source jar exists in local maven repositories, the debugger will automatically open the class source well. For your case, looks like the source jar is not ready (might be in downloading), Java extension just decompiles a stub code with method signature first for use.  You can wait for a moment and let Java extension finish downloading the source jar on backend first, because Java extension lazily downloads source jar when the source is needed.

**Reply by tap90** [reply]
The source java exists and it was correctly downloading. Maybe is there a problem if I'm not the administrator of the machine where run the application?

If I do > go to implementation > attach source

Nothing happen

**Reply by tap90** [reply]
I have solved by doing this:

set settings.json in vscode, "java.eclipse.downloadSources": true,"java.maven.downloadSources": true

**Reply by dudeNumber4** [reply]
My jar is local; waiting does nothing.  As @tap90 said, "attach source" does nothing.  Settings suggestion mentioned above does nothing.

**Comment by dudeNumber4** [comment, marked as answer]
Several upgrades of VS Code later (now at 1.74.1), the problem no longer exists.
